crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -300,4 +303,109 @@ impl BeaconState {
             signature: attestation.signature,
         })
     }
+
+    /// Increase the validator balance at index ``index`` by ``delta``.
+    pub fn increase_balance(&mut self, index: u64, delta: u64) {
+        if let Some(balance) = self.balances.get_mut(index as usize) {
+            *balance += delta;
+        }
+    }
+
+    /// Decrease the validator balance at index ``index`` by ``delta`` with underflow protection.
+    pub fn decrease_balance(mut self, index: u64, delta: u64) {
+        if let Some(balance) = self.balances.get_mut(index as usize) {
+            *balance -= delta;
```

Comment:
```suggestion
            balance.saturating_sub(delta)
```

the comment says underflow protection, I am assuming that would be a saturating sub?